core/src/main/scala/kafka/log/LogCleaner.scala

```diff
@@ -152,8 +152,8 @@ class LogCleaner(val config: CleanerConfig,
    * For testing, a way to know when work has completed. This method blocks until the
    * cleaner has processed up to the given offset on the specified topic/partition
    */
-  def awaitCleaned(topic: String, part: Int, offset: Long, timeout: Long = 30000L): Unit = {
-    while(!cleanerManager.allCleanerCheckpoints.contains(TopicAndPartition(topic, part)))
+  def awaitCleaned(topic: String, part: Int, offset: Long): Unit = {
```

Comment:
I have already run `LogCleanerIntegrationTest` and `DeleteTopicTest` a few times with infinite timeout and the tests run well. That is the reason why I think inifinite timeout works for existing tests. I don't think setting a default timeout value that is large enough for existing invocation in tests is a very good idea -- a default time value should ideally be good for most usecase from first principle.

But I agree with your point that we should never allow tests to run forever. How about I make the following change:

1) give default timeout value of 60*1000 ms. This is 4X the default `backoffMs` which is long enough for existing tests.

2) let `awaitCleaned` return a boolean value to indicate whether it has timed out or not -- this allows the tests to fail in case of timeout so that we can look into the problem.